### PR TITLE
feat: Enabled workload identity access for kms and gcs

### PR DESCRIPTION
**Pull request by amanpruthi**

### CODE REVIEW

**Comment by gls4** [comment]
Has this been tested against en existing cluster, as well as used to build a new cluster?

File: modules/app_gke/main.tf [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
   networking_mode = "VPC_NATIVE"
+  location        = var.location
 
   enable_intranode_visibility = true
 
+  # Conditionally enable workload identity
+  dynamic "workload_identity_config" {
+    for_each = var.workload_identity == true ? [1] : []
+    content {
+      workload_pool = "${local.project_id}.svc.id.goog"
```
Comment (gls4) [inline modules/app_gke/main.tf:21]: Typo?

**Review by gls4 — APPROVED** [review]
Working! :-)

**Comment by jsbroks** [comment]
@amanpruthi we are merge without security

**Pull request merged**